server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -2103,6 +2123,75 @@ public TSStatus createSchemaTemplate(TSCreateSchemaTemplateReq req) throws TExce
     }
   }
 
+  @Override
+  public TSStatus appendSchemaTemplate(TSAppendSchemaTemplateReq req) throws TException {
+    List<TSDataType> dataTypes = new ArrayList<>();
+    List<TSEncoding> encodings = new ArrayList<>();
+    List<CompressionType> compressionTypes = new ArrayList<>();
```

Comment:
use array TSDatatType[]
Also change the field type in Plan

Reply:
All plan maintains arrays rather than lists, but getters will return still return lists. Constructors support both list and array parameters.
interfaces in TSServiceImpl uses arrays now.